test/account_api_test.rb

```diff
@@ -108,6 +108,33 @@
     end
   end
 
+  describe "#delete_user_by_subject_identifier" do
+    let(:subject_identifier) { "the-subject-identifier" }
+    let(:path) { "/api/oidc-users/#{subject_identifier}" }
+
+    it "responds with 204 OK if the user existed and has been deleted" do
+      account_api
+        .given("there is a valid user session, and an existing user")
+        .upon_receiving("a delete-user request")
+        .with(method: :delete, path: path, headers: headers_with_json_body)
+        .will_respond_with(status: 204, headers: json_response_headers)
+
+      api_client.delete_user_by_subject_identifier(subject_identifier: subject_identifier)
+    end
+
+    it "responds with 404 if the user does not exist" do
+      account_api
+        .given("there is a valid user session, and the user cannot be found")
```

Comment:
This doesn't need a `given`, because (1) it doesn't need a valid user session, and (2) the default state is the user not existing.

It also doesn't need `headers_with_json_body` (only `headers`) because there is no response body.